provider/datagen/src/bin/datagen.rs

```diff
@@ -320,10 +320,10 @@ fn main() -> eyre::Result<()> {
                 .map(PathBuf::from)
                 .ok_or_else(|| eyre::eyre!("--out must be specified for --format=dir"))?,
             serializer: match matches.value_of("SYNTAX") {
-                Some("bincode") => Box::new(bincode::Serializer::default()),
-                Some("postcard") => Box::new(postcard::Serializer::default()),
+                Some("bincode") => Box::<bincode::Serializer>::default(),
```

Comment:
Interesting, I don't think adding turbofishes is great

Reply:
FWIW: I actually like it. It took me a bit to get used to reading it, but now I find it more readable than new+default.